go/vt/mysqlctl/azblobbackupstorage/azblob.go

```diff
@@ -40,26 +41,59 @@ import (
+			FlagName: "azblob_backup_account_name",
+		},
+	)
 
 	// This is the private access key
-	accountKeyFile string
+	accountKeyFile = viperutil.Configure(
+		configKey("account.key_file"),
+		viperutil.Options[string]{
+			FlagName: "azblob_backup_account_key_file",
+		},
+	)
 
 	// This is the name of the container that will store the backups
-	containerName string
+	containerName = viperutil.Configure(
+		configKey("container_name"),
+		viperutil.Options[string]{
+			FlagName: "azblob_backup_container_name",
+		},
+	)
 
 	// This is an optional prefix to prepend to all files
-	storageRoot string
+	storageRoot = viperutil.Configure(
+		configKey("storage_root"),
+		viperutil.Options[string]{
+			FlagName: "azblob_backup_storage_root",
+		},
+	)
 
-	azBlobParallelism int
+	azBlobParallelism = viperutil.Configure(
+		configKey("parallelism"),
+		viperutil.Options[int]{
+			Default:  1,
+			FlagName: "azblob_backup_parallelism",
+		},
+	)
 )
 
+const configKeyPrefix = "backup.storage.azblob"
+
+var configKey = viperutil.KeyPrefixFunc(configKeyPrefix)
+
 func registerFlags(fs *pflag.FlagSet) {
-	fs.StringVar(&accountName, "azblob_backup_account_name", "", "Azure Storage Account name for backups; if this flag is unset, the environment variable VT_AZBLOB_ACCOUNT_NAME will be used.")
-	fs.StringVar(&accountKeyFile, "azblob_backup_account_key_file", "", "Path to a file containing the Azure Storage account key; if this flag is unset, the environment variable VT_AZBLOB_ACCOUNT_KEY will be used as the key itself (NOT a file path).")
-	fs.StringVar(&containerName, "azblob_backup_container_name", "", "Azure Blob Container Name.")
-	fs.StringVar(&storageRoot, "azblob_backup_storage_root", "", "Root prefix for all backup-related Azure Blobs; this should exclude both initial and trailing '/' (e.g. just 'a/b' not '/a/b/').")
-	fs.IntVar(&azBlobParallelism, "azblob_backup_parallelism", 1, "Azure Blob operation parallelism (requires extra memory when increased).")
+	fs.String("azblob_backup_account_name", accountName.Default(), "Azure Storage Account name for backups; if this flag is unset, the environment variable VT_AZBLOB_ACCOUNT_NAME will be used.")
+	fs.String("azblob_backup_account_key_file", accountKeyFile.Default(), "Path to a file containing the Azure Storage account key; if this flag is unset, the environment variable VT_AZBLOB_ACCOUNT_KEY will be used as the key itself (NOT a file path).")
+	fs.String("azblob_backup_container_name", containerName.Default(), "Azure Blob Container Name.")
+	fs.String("azblob_backup_storage_root", storageRoot.Default(), "Root prefix for all backup-related Azure Blobs; this should exclude both initial and trailing '/' (e.g. just 'a/b' not '/a/b/').")
+	fs.Int("azblob_backup_parallelism", azBlobParallelism.Default(), "Azure Blob operation parallelism (requires extra memory when increased).")
+
+	viperutil.BindFlags(fs, accountName, accountKeyFile, containerName, storageRoot, azBlobParallelism)
```

Comment:
You mention in the doc that `BindFlags` should be called _after_ registering all flags and `Configure` is _usually_ called in `var` blocks. I assume we can still call `Configure` outside `var` blocks but before calling `BindFlags`?

Reply:
yup, you could equally do something like:

```go
var myValue viperutil.Value[int]

func init() {
    myValue = viperutil.Configure(...)

    servenv.OnParseFor("mycmd", func(fs *pflag.FlagSet) {
        // flags go here

        viperutil.BindFlags(fs, myValue)
    }
}
```